project/src/func.cpp

```diff
@@ -209,6 +216,27 @@ google::protobuf::Any *Func::warbleEvent(Database &kv_client_,
   return any;
 }
 
+std::unordered_set<std::string> Func::parseTagsFromWarbleText(const std::string& text) {
+  std::unordered_set<std::string> tags;
+  std::regex r("\\s+");
+  std::string sentence = std::regex_replace(text, r, std::string(1, char(176)));
+  std::stringstream ss(sentence);
+  std::string tag;
+  while (getline(ss, tag, char(176))) {
+    if (tag[0] == '#') {
+      tags.insert(tag);
+    }
+  }
+  return tags;
+}
+
+void Func::addHashtagToKvStore(warble::WarbleRequest &request, Database &kv_client_, int id) {
+  auto tags = parseTagsFromWarbleText(request.text());
+  for (const auto& tag: tags) {
+    kv_client_.put(tag, std::to_string(id));
```

Comment:
kv_store's put adds the value to a std::vector<std::string> so this would eventually be expensive in terms of storage but for the purposes of this function, looks fine to me.

Reply:
I feel it's better to keep unit tests in the same branch so I do not have a new branch. Could you spare a few minutes to take a look at the 3rd commit? Since the usage of --stream is much more end to end test than simply unit test, my unit test for stream event covers just one round.

Reply:
Yep, looks good to me. Go ahead and merge. Thanks